model-crd/dynamo-model-crd.md

```diff
@@ -0,0 +1,342 @@
+# DynamoModel: Kubernetes Custom Resource to simplify Model Lifecycle Management UX
+
+**Status**: In-review
+
+**Authors**: [biswapanda](https://github.com/biswapanda)
+
+**Category**: Architecture
+
+**Required Reviewers**: [Maksim, Itay, Anish, Ganesh, Neelay, Kavin]
+
+**Review Date**: [targeted: Oct 9, 2025]
+
+
+**Slack thread**: [link](https://nvidia.slack.com/archives/C06850J381Y/p1758647954211439?thread_ts=1758613631.539669&cid=C06850J381Y)
+
+# Summary
+
+This proposal introduces `DynamoModel`, a dedicated Kubernetes Custom Resource (CR) for managing model lifecycle in the Dynamo ecosystem. DynamoModel decouples model downloading, versioning, and caching from DynamoGraphDeployment (DGD), enabling consistent model references across deployments, benchmarks, and services while eliminating boilerplate code and preventing model version drift.
+
+# Motivation
+
+Currently, Dynamo users face three critical challenges:
+
+1. **Model Version Drift**: Inconsistent behavior occurs when AI-perf benchmarks use different model versions than deployments. This was observed during 70B model benchmarking where the deployment used stale weights while the benchmark job pulled the latest commit from HuggingFace.
```

Comment:
question: when we say the benchmark job used the latest - is this referencing the tokenizer for ai perf for example mismatched with the weights of the deployment? Not sure I follow what 'drift' means here - maybe 'mismatch' is a term - where different components can have different versions of the model?